cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -1474,9 +1480,8 @@ obtainDecodeEpochStateConstraints ShelleyBasedEraBabbage f = f
 
 -- | Return the slots at which a particular stake pool operator is
 -- expected to mint a block.
-currentEpochEligibleLeadershipSlots
-  :: forall era ledgerera .
-     ShelleyLedgerEra era ~ ledgerera
+currentEpochEligibleLeadershipSlots :: forall era ledgerera. ()
```

Comment:
Same here, why `()`?

Reply:
The reason I put the empty constraint `()` there is because it allows me to begin the next line with a fat arrow.

Code without `()`:

```
currentEpochEligibleLeadershipSlots :: forall era ledgerera.
     ShelleyLedgerEra era ~ ledgerera
  => Ledger.Era ledgerera
```

Code with `()`:

```
currentEpochEligibleLeadershipSlots :: forall era ledgerera. ()
  => ShelleyLedgerEra era ~ ledgerera
  => Ledger.Era ledgerera
```

The second is a little bit more `git` friendly because if I delete the first non-empty constraint, the git sees fewer lines changed.

For example:

```
diff --git a/cardano-api/src/Cardano/Api/LedgerState.hs b/cardano-api/src/Cardano/Api/LedgerState.hs
index bf0674fd1..7115b4a89 100644
--- a/cardano-api/src/Cardano/Api/LedgerState.hs
+++ b/cardano-api/src/Cardano/Api/LedgerState.hs
@@ -1494,7 +1494,6 @@ obtainDecodeEpochStateConstraints ShelleyBasedEraBabbage f = f
 -- | Return the slots at which a particular stake pool operator is
 -- expected to mint a block.
 currentEpochEligibleLeadershipSlots :: forall era ledgerera. ()
-  => ShelleyLedgerEra era ~ ledgerera
   => Ledger.Era ledgerera
   => Consensus.PraosProtocolSupportsNode (Api.ConsensusProtocol era)
   => HasField "_d" (Core.PParams ledgerera) UnitInterval
```

vs

```
diff --git a/cardano-api/src/Cardano/Api/LedgerState.hs b/cardano-api/src/Cardano/Api/LedgerState.hs
index 493594e8b..0be10b221 100644
--- a/cardano-api/src/Cardano/Api/LedgerState.hs
+++ b/cardano-api/src/Cardano/Api/LedgerState.hs
@@ -1494,8 +1494,7 @@ obtainDecodeEpochStateConstraints ShelleyBasedEraBabbage f = f
 -- | Return the slots at which a particular stake pool operator is
 -- expected to mint a block.
 currentEpochEligibleLeadershipSlots :: forall era ledgerera.
-     ShelleyLedgerEra era ~ ledgerera
-  => Ledger.Era ledgerera
+     Ledger.Era ledgerera
   => Consensus.PraosProtocolSupportsNode (Api.ConsensusProtocol era)
   => HasField "_d" (Core.PParams ledgerera) UnitInterval
   -- => Crypto.Signable (Crypto.VRF (Ledger.Crypto ledgerera)) Shelley.Spec.Seed
```

Reply:
Fair point. We can adopt this style :+1: